src/Bicep.Core/Syntax/ResourceDeclarationSyntax.cs

```diff
@@ -38,6 +38,9 @@ public ResourceDeclarationSyntax(Token keyword, IdentifierSyntax name, SyntaxBas
 
         public SyntaxBase Body { get; }
 
+        public SyntaxBase ResolvedBody =>
+            this.Body is IfExpressionSyntax conditionalBody ? conditionalBody.ConsequenceExpression : this.Body;
```

Comment:
Does this require a grammar update? (`grammar.md`).

If I'm understanding correctly, looks like with these changes we've got something like:
```
resourcedecl -> "resource" identifier string "=" body
body -> conditionalBody | object
conditionalBody -> "if" "(" expression ")" object
```

Just wonder, rather than using recursion for the definition, would it make sense to model the grammar as:
```
resourcedecl -> "resource" identifier string = ifexpr? object
ifexpr -> "if" "(" expression ")"
```

This is similar to what we did for the param default syntax, and might make chaining (when we have loops) and recursion easier to deal with. I think it also removes the need for the `ResolvedBody` property.

Reply:
Yeah, I think it makes sense to not use recursion for the definition, and I like how it simplifies things a lot. The recursion version may work better if we are going to add an alternative expression to support `else`, but I doubt that's necessary, since `if...else...` can be replaced with a ternary operator. I'm going to change the grammar to the suggested version.

Reply:
Updated the grammar to not sue recursion. I changed `IfExpressionSyntax` to `IfCondtionSyntax` though, because it's not an expression anymore.